src/remoted/ar-forward.c

```diff
@@ -137,17 +137,37 @@ void *AR_Forward(__attribute__((unused)) void *arg)
 
             /* Send to a pre-defined agent */
             else if (ar_location & SPECIFIC_AGENT) {
-                ar_location++;
+                char *org_agent_list = NULL;
+                char *agent_list = NULL;
+                char *token = NULL;
+                int  agent_id_err = 0;
 
-                agent_id = OS_IsAllowedID(&keys, ar_agent_id);
+                ar_location++;
 
-                if (agent_id < 0) {
+                org_agent_list = agent_list = strdup(ar_agent_id);
+                //verbose("%s: [INFO] selected agent:=%s", ARGV0, org_agent_list);
+                if (org_agent_list == NULL) {
                     key_unlock();
                     merror(AR_NOAGENT_ERROR, ARGV0, ar_agent_id);
-                    continue;
+                    continue ;
                 }
-
-                send_msg((unsigned)agent_id, msg_to_send);
+                while ((token = strsep(&agent_list, ",")) != NULL) {
```

Comment:
What I've found online might be old information, but it doesn't look like Solaris has strsep. I didn't try to check on AIX/HP-UX.